core/dbt/compilation.py

```diff
@@ -482,7 +483,13 @@ def compile(self, manifest: Manifest, write=True, add_test_edges=False) -> Graph
 
         if write:
             self.write_graph_file(linker, manifest)
-        print_compile_stats(stats)
+
+        # Only in the case the object is specifically a ListTask do we avoid printing these stats.
+        try:
+            if not isinstance(self.config.args.get("cls", None), list_task.ListTask):
+                print_compile_stats(stats)
+        except AttributeError:
+            print_compile_stats(stats)
```

Comment:
It's not clear to me what you're trying to do here. If the "try" of doing print_compile_stats fails you do it again?

Reply:
Read another way: only print compile stats if the task is ListTask. If the task object is None (sometimes this happens), the `get` will fail. In that case, still print the stats.

Reply:
I don't think the try/except serves any purpose here.  Wouldn't the "if not... / print_compile_stats" be enough?

Reply:
We need the try catch guard between the `if not` can throw an `AttributeError` if a `self.config.args` equal to `None` is passed in, which happens numerous times through execution. I'll double check though :)

Reply:
Just double checked and if I remove the try-catch, this is what happens:
<img width="1207" alt="image" src="https://user-images.githubusercontent.com/67295367/213606525-4adc4ae9-90ee-4108-97c5-25c71f4f8fff.png">

If I have the try-catch in place, that behavior doesn't happen
<img width="1508" alt="image" src="https://user-images.githubusercontent.com/67295367/213606446-d99ec435-fee3-4c74-b903-a0299d598468.png">

There could be a more elegant way to observe this is a ListTask but with the `self.config.args` object, we have to worry about `None` for `args`.

Reply:
I went through all this only to find out the underlying object has changed to a `Namespace`. Pardon me, under this setup, what you're saying is right. My changes were developed before merging main. I'll fix this up 👍